### PR TITLE
Format Script Update

### PR DESCRIPTION
**Pull request by hensldm**
Brings over the new `format.py` from OoT that actually checks you have `clang-tidy`, along with some new clang-tidy options to force function definitions and declarations to be consistent. Also ran it across the whole repo.

### CODE REVIEW

**Review by jpburnett — APPROVED** [review]
Started reviewing thinking that this was just formatting changes, but saw some naming fixes. Just checking to see if this was intended?

**Comment by hensldm** [comment]
Yeah, most of the renaming come from `clang-tidy` itself, but I did make two intentional changes that I noticed
- `void EnItem00_DrawHeartContainer(EnItem00* actor, PlayState* play)` -> `void EnItem00_DrawHeartContainer(EnItem00* this, PlayState* play)`
- `void ObjRaillift_UpdatePosition(ObjRaillift* this, s32 idx)` -> `void ObjRaillift_UpdatePosition(ObjRaillift* this, s32 index)`

**Review by AngheloAlf — APPROVED** [review]

**Comment by hensldm** [comment]
Going to draft this. Want to to see what happens with https://github.com/zeldaret/oot/pull/1331 and if we should ~~steal~~, I mean ask nicely if we can use it.

**Comment by hensldm** [comment]
Okay this should be good to go again. Would recommend just trying out the new script if you haven't already.

File: Jenkinsfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ pipeline {
         stage('Check formatting') {
             steps {
                 echo 'Checking formatting...'
-                sh 'bash -c "tools/check_format.sh 2>&1 >(tee tools/check_format.txt)"'
+                sh 'bash -c "tools/check_format.sh -j8 2>&1 >(tee tools/check_format.txt)"'
```
Comment (hensldm) [inline Jenkinsfile:14]: To fix the check_format script I've currently just hardcoded it to 8 jobs. Should this just be dependent on `nproc` or what?

Comment (Roman971) [reply Jenkinsfile:14]: I recommend just having `check_format.sh` call `format.py` with `-j` on its own, without specifying a number of jobs, and format.py should default to the number of CPUs for the machine

**Comment by AngheloAlf** [comment]
Is this PR still waiting for the agents to be updated?

**Comment by AngheloAlf** [comment]
Jenkins is complaining there are unformatted files, but I can't reproduce the problem locally. Shouldn't this be using a specific clang format version?

**Comment by hensldm** [comment]
> Jenkins is complaining there are unformatted files, but I can't reproduce the problem locally. Shouldn't this be using a specific clang format version?

Hmm, locally I see EnZos change due to clang-tidy (1 function declaration and define don't match). Curious what you did locally (though probably best to debug in discord). As for the other file, it looks like a core file??? That is weird.

**Comment by hensldm** [comment]
Converting to draft again. For some reason Slug is coring when trying to run clang-apply-replacements, and would like to figure that out first.

**Comment by Roman971** [comment]
FYI I made a PR with some fixes to format.py in oot (see zeldaret/oot#1387), and it will probably prevent that coring issue you ran into

**Comment by hensldm** [comment]
Undrafted. Looks good again. Thanks Roman.

**Comment by EllipticEllipsis** [comment]
On Mac I had to add symlinks for `clang-tidy` and `clang-apply-replacements` (after installing all of llvm, unfortunately `clang-tidy` does not have its own distribution on homebrew), so we should add that to the tools document or the script's help.

```
$ brew install llvm
$ ln -s "$(brew --prefix llvm)/bin/clang-tidy" "/usr/local/bin/clang-tidy"
$ ln -s "$(brew --prefix llvm)/bin/clang-apply-replacements" "/usr/local/bin/clang-apply-replacements"
```
was the right thing according to the instructions I followed.

**Comment by EllipticEllipsis** [comment]
And I suppose I should note I'm on 15.0.0 and it didn't change anything.

File: include/functions.h [hunk 2]
```diff
@@ -5,25 +5,26 @@
...
+s32 DmaMgr_DmaHandler(OSPiHandle* pihandle, OSIoMesg* mb, s32 direction);
+DmaEntry* DmaMgr_FindDmaEntry(u32 vrom);
+u32 DmaMgr_TranslateVromToRom(u32 vrom);
+s32 DmaMgr_FindDmaIndex(u32 vrom);
 const char* func_800809F4(u32 param_1);
 void DmaMgr_ProcessMsg(DmaRequest* req);
 void DmaMgr_ThreadEntry(void* arg);
-s32 DmaMgr_SendRequestImpl(DmaRequest* request, void* vramStart, uintptr_t vromStart, size_t size, UNK_TYPE4 unused, OSMesgQueue* callback, void* callbackMesg);
+s32 DmaMgr_SendRequestImpl(DmaRequest* request, void* vramStart, uintptr_t vromStart, size_t size, UNK_TYPE4 unused,
+                           OSMesgQueue* queue, void* msg);
```
[the remark this answers is not in the document]
Comment (hensldm) [reply include/functions.h:19]: Note, I removed the clang-apply-replacements opts that was causing this (which differs from OoT)

**Review by EllipticEllipsis — APPROVED** [review]

**Pull request merged**